### PR TITLE
[flutter_tools] reduce globals in web validator and chrome launcher

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Towards simpler tool testing

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/web/web_validator_test.dart [hunk 1]
```diff
@@ -10,56 +12,63 @@ import 'package:process/process.dart';
...
+      chromeLauncher: chromeLauncher,
+      fileSystem: fileSystem,
+    );
+  });
 
-    test('Can find macOS executable ', () => testbed.run(() async {
-      final ValidationResult result = await webValidator.validate();
-      expect(result.type, ValidationType.installed);
-    }));
+  test('WebValidator can find executable on macOS', () async {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/web/web_validator_test.dart]: can this be `testWithoutContext()`?

**Review by christopherfujino — APPROVED** [review]
LGTM, with one nit question.

File: packages/flutter_tools/test/general.shard/web/web_validator_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/web/web_validator_test.dart]: Ahh good catch, fixed

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**